### DISCUSSION TITLE
How to Filter by Relationship?

### DISCUSSION THREAD

**Discussion by HeronCS** [Budibase/budibase#5661] [Help] [answered]
Within a table, how to filter by relationship?

Unable to use define filter, as the relationships don't appear.

**Comment by melohagan** [comment]
Hey @HeronCS 

Unfortunately filter by relationship is not possible. Feature request here: https://github.com/Budibase/budibase/issues/5585

**Comment by oitkenmen** [comment]
Hello, when user saves the form, you can add additional columns related to that user (e.g. username or other fields). They will be created automatically. Then, you can filter data by these additional columns.

**Comment by YTDcontrol** [comment]
Hi. You can write a query with those relationships also. Then you can add those columns gererated by the query to the dynamic filter.

**Reply by HeronCS** [reply]
Could you explain this further, please.

**Comment by magickaito** [comment, marked as answer]
I am using a work around - using static formula. 

E.g. I have Project tables and Project Issues table. 

Every project issue has a relationship column "Project" that links to a project.

I want to filter Project Issues by Project.

In Project Issue table, I create a static formula column "Project ID" that derives value from the linked Project relationship. 

`return $("Project")[0]._id`

Now I can filter the project issues to only show rows where its "Project ID" column matches the selected project ID in the option picker.

Example:
[filter-export-1655351132059.txt](https://github.com/Budibase/budibase/files/8915477/filter-export-1655351132059.txt)

**Reply by oitkenmen** [reply]
It's a good solution. It works great with static values. But I've noticed it causes a "Document Update Conflict" error with dynamic values.

**Reply by HeronCS** [reply]
a dynamic value as in a column with a formula?

**Reply by oitkenmen** [reply]
a formula from formula

**Reply by melohagan** [reply]
@HeronCS They mean a static formula like so:
![Screenshot 2022-06-15 at 14 39 33](https://user-images.githubusercontent.com/101575380/173841489-e9210bdf-2a06-467f-88a5-ebfc739d2ee4.png)

@magickaito This is a great workaround! Thank you very much for sharing :)

**Reply by magickaito** [reply]
@melohagan u r welcome!

**Reply by HeronCS** [reply]
Let me try this out.
